schema.graphql

```diff
@@ -1,18 +1,117 @@
-type BackerStakingHistory @entity {
+type BackerStakingHistory @entity(immutable: false) {
   id: Bytes!
-  backerTotalAllocation_: BigInt!
-  accumulatedTime_: BigInt!
-  lastBlockNumber_: BigInt!
-  lastBlockTimestamp_: BigInt!
-  gauges_: [GaugeStakingHistory!]! @derivedFrom(field: "backer_")
+  backerTotalAllocation: BigInt!
+  accumulatedTime: BigInt!
+  lastBlockNumber: BigInt!
+  lastBlockTimestamp: BigInt!
+  gauges: [GaugeStakingHistory!]! @derivedFrom(field: "backer")
 }
 
-type GaugeStakingHistory @entity {
+type GaugeStakingHistory @entity(immutable: false) {
   id: Bytes!
-  gauge_: Bytes! # address
-  backer_: BackerStakingHistory!
-  allocation_: BigInt!
-  accumulatedAllocationsTime_: BigInt!
-  lastBlockNumber_: BigInt!
-  lastBlockTimestamp_: BigInt!
+  gauge: Bytes! # address
+  backer: BackerStakingHistory!
+  allocation: BigInt!
+  accumulatedAllocationsTime: BigInt!
+  lastBlockNumber: BigInt!
+  lastBlockTimestamp: BigInt!
 }
+
+type Builder @entity(immutable: false) {
+  id: Bytes!
+  gauge: Bytes!
+  backerRewardPercentage: BigInt!
+  rewardReceiver: Bytes!
+  state: BuilderState! @derivedFrom(field: "builder")
+  isHalted: Boolean!
+  lastCycleRewards: BigInt!
+  totalAllocation: BigInt!
+  rewardShares: BigInt!
+  proposals: [Proposal!]! @derivedFrom(field: "builder")
+  rewardsClaimed: [BuilderRewardsClaimed!]! @derivedFrom(field: "builder")
+}
```

Comment:
could we also calculate `estimatedBackerRewardsPct`?
As we need `builder.rewardShares`, `builder.backerRewardPercentage` and `cycle.totalPotentialReward` for the calculation, and check the state against:

```
(
    stateFlags &&
    stateFlags.activated &&
    stateFlags.communityApproved &&
    stateFlags.kycApproved &&
    !stateFlags.revoked
  )
```

Reply:
Yes, I think we could. Let's decide together if we can do it in time for the current release